File: app/src/main/java/dagger/nambv/app/demo/di/module/AppModule.java
package dagger.nambv.app.demo.di.module;

import android.app.Application;
import android.content.res.Resources;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;
import dagger.nambv.app.demo.BaseApplication;

@Module
public class AppModule {
    BaseApplication app;

    public AppModule(BaseApplication app) {
        this.app = app;
    }

    @Provides
    @Singleton
    Application provideApplication() { return app; }

    @Provides
    @Singleton
    Resources provideResources() {
        return app.getResources();
    }
}

File: app/src/main/java/dagger/nambv/app/demo/di/module/AppModule.kt
package dagger.nambv.app.demo.di.module

import android.app.Application
import android.content.res.Resources
import dagger.Module
import dagger.Provides
import dagger.nambv.app.demo.BaseApplication
import javax.inject.Singleton

@Module
class AppModule(var app: BaseApplication) {

    @Provides
    @Singleton
    fun provideApplication(): Application {
        return app
    }

    @Provides
    @Singleton
    fun provideResources(): Resources {
        return app.resources
    }
}
